activesupport/lib/active_support/digest.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module ActiveSupport
+  class Digest
+    class <<self
+      attr_reader :hash_digest_class
+
+      def hash_digest_class=(clazz)
+        raise ArgumentError, "#{clazz} is expected to implement hexdigest class method" unless clazz.respond_to?(:hexdigest)
+        @hash_digest_class = clazz
+      end
+
+      def hexdigest(arg)
+        new.hexdigest(arg)
+      end
+    end
+
+    def initialize(digest_class: nil)
+      @digest_class = digest_class
+    end
+
+    def hexdigest(arg)
+      if @digest_class || self.class.hash_digest_class
+        (@digest_class || self.class.hash_digest_class).hexdigest(arg)
+      else
+        ::Digest::MD5.hexdigest(arg)
+      end
```

Comment:
This conditional could be just `(@digest_class || self.class.hash_digest_class || ::Digest::MD5).hexdigest(arg)`